From 93b9a179119394395c72e62e59a73d29e9bba735 Mon Sep 17 00:00:00 2001
From: Quirin Gylstorff <quirin.gylstorff@siemens.com>
Date: Mon, 7 Feb 2022 09:28:39 +0100
Subject: [PATCH 8/8] debian: Remove SWUpdate USB service and Udev rules

The current implementation will install an abitrary SWUpdate binary
from a plug-in USB stick. This is a major security risk for devices
using the SWUpdate package from Debian.

Remove the installation and the files from the debian folder.

Signed-off-by: Quirin Gylstorff <quirin.gylstorff@siemens.com>
---
 debian/rules                          | 1 -
 debian/swupdate.swupdate-usb@.service | 8 --------
 debian/swupdate.udev                  | 2 --
 3 files changed, 11 deletions(-)
 delete mode 100644 debian/swupdate.swupdate-usb@.service
 delete mode 100644 debian/swupdate.udev

diff --git a/debian/rules b/debian/rules
index e1c4a921..84ed55d4 100755
--- a/debian/rules
+++ b/debian/rules
@@ -103,7 +103,6 @@ override_dh_auto_install:
 override_dh_installsystemd:
 	dh_installsystemd --no-start
 	dh_installsystemd --name=swupdate-progress
-	dh_installsystemd --no-start --name=swupdate-usb@
 
 ifeq (,$(filter pkg.swupdate.bpo,$(DEB_BUILD_PROFILES)))
 override_dh_gencontrol:
diff --git a/debian/swupdate.swupdate-usb@.service b/debian/swupdate.swupdate-usb@.service
deleted file mode 100644
index eda9d153..00000000
--- a/debian/swupdate.swupdate-usb@.service
+++ /dev/null
@@ -1,8 +0,0 @@
-[Unit]
-Description=usb media swupdate service
-Requires=swupdate-progress.service
-
-[Service]
-ExecStartPre=/bin/mount /dev/%I /mnt
-ExecStart=/bin/sh -c "swupdate-client -v /mnt/*.swu"
-ExecStopPost=/bin/umount /mnt
diff --git a/debian/swupdate.udev b/debian/swupdate.udev
deleted file mode 100644
index b4efd0b7..00000000
--- a/debian/swupdate.udev
+++ /dev/null
@@ -1,2 +0,0 @@
-ACTION=="add", KERNEL=="sd*", SUBSYSTEM=="block", ENV{ID_BUS}=="usb", ENV{ID_FS_USAGE}=="filesystem", TAG+="systemd", ENV{SYSTEMD_WANTS}+="swupdate-usb@%k.service"
-
-- 
2.34.1

